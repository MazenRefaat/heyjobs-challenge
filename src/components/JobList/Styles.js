import styled from 'styled-components';

export const StyledList = styled.section`
    height: 400px;
    overflow: scroll;
    border: 1px solid rgba(230, 230, 230, 0.5);
`;