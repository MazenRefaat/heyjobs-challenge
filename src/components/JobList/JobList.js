import React, { useState, useEffect } from 'react';
import { JobsListService } from '../../services/JobsService';
import Card from '../../components/Card/Card';
import { StyledList } from './Styles';

/**
 * JobList 
 * @type Component
 * @description responsible for fetching and listing available jobs 
 * @return List of all available jobs 
 */

const JobList = () => {
    const [availableJobs, setAvailableJobs] = useState([]);

    useEffect(()=>{
        JobsListService().then(res => {
            setAvailableJobs(res.data)
        }).catch(err => {
            console.log('error', err)
        })
    },[])

    return (
        <StyledList>
            {
                availableJobs.map((job, key)=> (
                    <Card key={key} item={job} />
                ))
            }
        </StyledList>
    )
}


export default JobList;