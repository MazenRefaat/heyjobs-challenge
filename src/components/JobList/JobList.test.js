import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import JobList from './JobList';

configure({ adapter: new Adapter() });

describe('<JobList/>', ()=> {
    let wrapper = mount(<JobList />);
    const setJobsState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation((jobsState) => [jobsState, setJobsState]);

    beforeEach(()=> {
        wrapper = mount(<JobList />);
    })

    afterEach(()=> {
        jest.clearAllMocks();
    })

    it('should match snapshot', ()=> {
        expect(wrapper).toMatchSnapshot();
    })

    it('should update availableJobs state on API fetch of jobs', ()=> {
        const fetchResponse = {
            data: 'test'
        }

        global.fetch = jest.fn().mockImplementation(()=> 
            Promise.resolve(mockResponse(200, null, JSON.stringify(fetchResponse)))
        );

        setJobsState(fetchResponse.data)
        expect(setJobsState).toHaveBeenCalledWith(fetchResponse.data);
    })
})