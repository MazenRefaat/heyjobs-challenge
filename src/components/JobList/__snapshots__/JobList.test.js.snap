// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobList/> should match snapshot 1`] = `ReactWrapper {}`;
